components/ExerciseCard/ExerciseCard.test.tsx

```diff
@@ -3,45 +3,93 @@ import React from 'react'
 import { fireEvent, render } from '@testing-library/react'
 import ExerciseCard from './ExerciseCard'
 
-describe('ExerciseCard component', () => {
-  it('Should render an exercise card', () => {
-    const exampleProblem = `let a = 5
+const successMessage = '🎉 Your answer is correct!'
+const errorMessage = 'Your answer is incorrect - please try again.'
+
+const exampleProblem = `let a = 5
 a = a + 10
 // what is a?`
-    const exampleAnswer = '15'
-    const exampleExplanation = `You can reassign variables that are initialized with "let".`
+const exampleAnswer = '15'
+const exampleExplanation = `You can reassign variables that are initialized with "let".`
 
-    const { getByRole, queryByText, getByLabelText } = render(
+describe('ExerciseCard component', () => {
+  it('Should render an exercise card', async () => {
+    const setAnswerShown = jest.fn()
+    const setMessageKey = jest.fn()
+
+    const { getByRole, queryByText } = render(
       <ExerciseCard
         problem={exampleProblem}
         answer={exampleAnswer}
         explanation={exampleExplanation}
+        answerShown={false}
+        setAnswerShown={setAnswerShown}
+        messageKey={'EMPTY'}
+        setMessageKey={setMessageKey}
       />
     )
 
     // Test that an error message shows if the user is wrong
 
-    const errorMessage = 'Your answer is incorrect - please try again.'
-
     expect(queryByText(errorMessage)).not.toBeInTheDocument()
 
     const submitButton = getByRole('button', { name: 'SUBMIT' })
     fireEvent.click(submitButton)
 
-    expect(queryByText(errorMessage)).toBeInTheDocument()
+    expect(setAnswerShown.mock.calls).toEqual([])
+    expect(setMessageKey.mock.calls).toEqual([['ERROR']])
```

Comment:
Info

I think a shorthand is `expect(setMessageKey).toBeCalledWith('ERROR')`. Not sure about how it'll work for line 30.

Reply:
Sounds good.

Reply:
Done.